load_ali.m

```diff
@@ -1,105 +1,102 @@
 function [Key,Basic,Pdf,Phone,Phone_seq] = load_ali(alifile,model)
-% Load from a Kaldi algignment file in integer basic, pdf, and 
-% phone formats, using decoding information from model.
+  % Load from a Kaldi algignment file in integer basic, pdf, and 
+  % phone formats, using decoding information from model.
+  %
+  % OUTPUTS:
+  % Key: cell array of key/unique-identifiers
+  % Basic: cell array of transition-id vectors
+  % Pdf: cell array of pdf-id vectors
+  % Phone: cell array of phone-id vectors
+  % Phone_seq: cell array of [phone-ids; lengths] matrices
 
-% Default argument.  This pdf alignment file has 129 lines.
-if nargin < 1
+  if nargin < 2
     alifile = '/projects/speech/sys/kaldi-trunk/egs/rm/s5/exp/mono/ali.1.gz';
     model = '/projects/speech/sys/kaldi-trunk/egs/rm/s5/exp/mono/final.mdl';
-end
+  end
 
-% Open input streams for alignments in various formats.
-% Basic alignment with transition IDs.
-basic_ali = '/tmp/align3_basic_ali.txt';
-cmd = ['gzcat ', alifile, ' > ', basic_ali];
-system(cmd);
-basic_stream = fopen(basic_ali);
+  % Open input streams for alignments in various formats.
+  % Basic alignment with transition IDs.
+  basic_ali = '/tmp/align3_basic_ali.txt';
+  cmd = ['gunzip -c ', alifile, ' > ', basic_ali];
```

Comment:
"gzcat" is equivalent to "gunzip -c". Ubuntu does not support "gzcat" but does support "gunzip", so this change should (hopefully) make this command more portable.